server/src/main/java/org/opensearch/cluster/metadata/QueryGroup.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.UUIDs;
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.search.ResourceType;
+import org.joda.time.Instant;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to define the QueryGroup schema
+ * {
+ *              "_id": "fafjafjkaf9ag8a9ga9g7ag0aagaga",
+ *              "resourceLimits": {
+ *                  "jvm": 0.4
+ *              },
+ *              "resiliency_mode": "enforced",
+ *              "name": "analytics",
+ *              "updatedAt": 4513232415
+ * }
+ */
+@PublicApi(since = "2.15")
```

Comment:
thinking these components should be marked instead with `@ExperimentalApi` until this is ready for a release?